### PR TITLE
Require clients to provide a redirect location

### LINKED ISSUE #44: Require providing the redirect location as request parameter
As an application developer, I'd like to provide the redirect location as request parameter, in order to be able to use couch_email_auth with multiple applications.

For security reasons, this should come along with a whitelist of domains that are allowed in redirect locations.

This implies getting rid of the `redirectLocation` configuration option.

### PR DESCRIPTION
**Pull request by KlausTrainer**
Closes #44.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -19,5 +19,18 @@ module.exports = function() {
...
+  var config = rc('couch_email_auth', defaultConfig);
+
+  if (typeof config.allowedRedirectHosts === 'string' &&
+      config.allowedRedirectHosts.trim() !== '') {
+    config.allowedRedirectHosts = config.allowedRedirectHosts.split(',').map(
+      function(host) {
+        return host.trim();
+      }
+    );
+  } else {
```
Comment (robertkowalski) [inline lib/config.js]: super minor and no blocker for a merge, but i would prefer early returns

**Comment by robertkowalski** [comment]
cool +1

File: lib/config.js (hunk 1 above)
Comment (KlausTrainer) [inline lib/config.js]: Do you have a concrete suggestion?

I'm +1 on early returns where they result in code paths that are easier to follow. But here I want to make sure that `config.allowedRedirectHosts` is always set and and of type array, and I don't see how to do an early return without making it less understandable.

Comment (robertkowalski) [inline lib/config.js]:
```js

  config.allowedRedirectHosts = [];
  if (typeof config.allowedRedirectHosts === 'string' &&
      config.allowedRedirectHosts.trim() !== '') {
    config.allowedRedirectHosts = config.allowedRedirectHosts.split(',').map(
      function(host) {
        return host.trim();
      }
    );
  } 
  return config;
```

not sure if it is easier to read, probably quite subjective :)

as I said, feel free to merge

Comment (robertkowalski) [inline lib/config.js]: aw wait, that makes no sense!

sorry, go ahead, nothing to see here ;)

Comment (KlausTrainer) [inline lib/config.js]: No problem :) Thanks again!

**Pull request closed without merging**